Improve `st.progress` UI

imjuangarcia opened the pull request:
## 📚 Context

This PR adjusts the UI of `st.progress` to match our [Core components pattern library](https://www.figma.com/file/fYUxW21y2iwWQ67gFZROUE/Core-Components?node-id=20746%3A181107)

- What kind of change does this PR introduce?

  - [ ] Bugfix
  - [ ] Feature
  - [ ] Refactoring
  - [x] Other, please describe: Design change

## 🧠 Description of Changes

- Update the default `SIZE` property, to make it smaller;
- Update border radius to make it more rounded;
- Update the fill background color.

  - [ ] This is a breaking API change
  - [x] This is a visible (user-facing) change

**Revised:**

![Screen Shot 2022-07-21 at 3 53 14 PM](https://user-images.githubusercontent.com/34423371/180294787-96bd02a2-0765-4b69-88aa-4d69d881092e.png)

![Screen Shot 2022-07-21 at 3 53 36 PM](https://user-images.githubusercontent.com/34423371/180294790-9fb80798-0e94-45c4-83bb-da04fc103a5f.png)

**Current:**

![progressbar-dark snap](https://user-images.githubusercontent.com/34423371/180294741-d3af3604-d311-448d-829c-e88136331550.png)

![progressbar snap](https://user-images.githubusercontent.com/34423371/180294747-cd74945a-0545-4ef6-87af-e0da014d4831.png)

## 🧪 Testing Done

- [x] Screenshots included
- [ ] Added/Updated unit tests
- [x] Added/Updated e2e tests

## 🌐 References

- https://www.notion.so/streamlit/st-progress-b87706732aa547d8b662a6d5d857b44e

---

**Contribution License Agreement**

By submitting this pull request you agree that all contributions to this project are made under the Apache 2.0 license.

sfc-gh-jgarcia committed:
Improve progress bar UI

sfc-gh-jgarcia committed:
Remove snapshots

sfc-gh-jgarcia committed:
Add progress bar snapshots

sfc-gh-jgarcia committed:
Remove other offending tests

sfc-gh-jgarcia committed:
Add hello and date_input snapshots

vdonato reviewed (approved):
LGTM!

One quick question before merging this -- do you happen to know if there's any difference between the old and new `hello` snapshots? I can't see anything by just looking at the two side by side.

I wonder if the tests just happened to fail without being a snapshot issue and the update wasn't necessary (I think the hello e2e tests are occasionally flaky). I guess if that's the case, it doesn't really matter either way if we update the snapshots or not, so this can be merged as-is.

imjuangarcia commented:
@vdonato Yeah, I was wondering about the same thing when I saw the failure... I didn't see any changes either when I updated them. We did change some spacing stuff on `st.selectbox` and `st.multiselect` [here](https://github.com/streamlit/streamlit/pull/4995), so maybe it's related to that? I updated the `hello` tests on that PR as well though, so I'm not really sure.

vdonato commented:
Looking at the cypress run from the commit that failed, I think it may have just been a fluke that it failed when you ran into it and the snapshot update may not have been needed. The updated snapshot does probably differ (very, very) slightly from the current one. This can happen since we do allow up to 1% differences between the snapshot taken and the one saved to prevent test failures from being too noisy.

I don't think whatever difference there is is large enough for a person to be able to tell, so I don't think we need to worry about it
